psst-gui/src/webapi/client.rs

```diff
@@ -213,6 +275,243 @@ impl WebApi {
+        pub struct Artists {
+            items: Vec<ArtistsItem>,
+        }
+        
+        #[derive(Deserialize)]
+        pub struct ArtistsItem {
+            profile: Profile,
+            uri: String,
+        }
+        
+        #[derive(Deserialize)]
+        pub struct Profile {
+            name: String,
+        }
+
+        #[derive(Deserialize)]
+        pub struct Attribute {
+            key: String,
+            value: String,
+        }
+    
+        #[derive(Deserialize)]
+        pub struct Images {
+            items: Vec<ImagesItem>,
+        }
+    
+        #[derive(Deserialize)]
+        #[serde(rename_all = "camelCase")]
+        pub struct ImagesItem {
+            extracted_colors: ExtractedColors,
+            sources: Vec<Source>,
+        }
+    
+        #[derive(Deserialize)]
+        #[serde(rename_all = "camelCase")]
+        pub struct ExtractedColors {
+            color_dark: ColorDark,
+        }
+    
+        #[derive(Deserialize)]
+        #[serde(rename_all = "camelCase")]
+        pub struct ColorDark {
+            hex: String,
+            is_fallback: bool,
+        }
+    
+        #[derive(Deserialize)]
+        pub struct Source {
+            height: Option<serde_json::Value>,
+            url: String,
+            width: Option<serde_json::Value>,
+        }
+    
+        #[derive(Deserialize)]
+        pub struct OwnerV2 {
+            data: OwnerV2Data,
+        }
+    
+        #[derive(Deserialize)]
+        pub struct OwnerV2Data {
+            #[serde(rename = "__typename")]
+            typename: String,
+            name: String,
+            uri: String,
+        }
+    
+        #[derive(Deserialize)]
+        #[serde(rename_all = "camelCase")]
+        pub struct PagingInfo {
+            next_offset: Option<serde_json::Value>,
+        }
+    
+        #[derive(Deserialize)]
+        pub struct Extensions {}
+
+         // Extract the playlists
+         let result: Welcome = self.load(request)?;
+        
+        let playlist: Vector<Playlist> = result.data.home_sections.sections
+        .iter()
+        .flat_map(|section| {
+            section.section_items.items.iter().map(|item| {
+                let uri = item.uri.clone();
+                let id = uri.split(':').last().unwrap_or("").to_string();
+
+                Playlist {
+                    id: id.into(),
+                    name: item.content.data.name.clone().unwrap_or_default().into(),
+                    images: item.content.data.images.as_ref().map(|images| 
+                        images.items.iter().map(|img| data::utils::Image {
+                            url: img.sources.first().map(|s| s.url.clone()).unwrap_or_default().into(),
+                            width: None,
+                            height: None,
+                        }).collect()
+                    ),
+                    description: item.content.data.description.clone().unwrap_or_default().into(),
```

Comment:
We can use `sanitize_html` to remove links from the descriptions. I'll add it as a commit.